homeassistant/components/sabnzbd.py

```diff
@@ -0,0 +1,258 @@
+        await async_configure_sabnzbd(hass, info, ssl)
+
+    discovery.async_listen(hass, SERVICE_SABNZBD, sabnzbd_discovered)
+
+    conf = config.get(DOMAIN)
+    if conf is not None:
+        use_ssl = conf.get(CONF_SSL)
+        name = conf.get(CONF_NAME)
+        api_key = conf.get(CONF_API_KEY)
+        await async_configure_sabnzbd(hass, conf, use_ssl, name, api_key)
+    return True
+
+
+@callback
+def async_setup_sabnzbd(hass, sab_api, config, name):
+    """Setup SABnzbd sensors and services."""
+    sab_api_data = SabnzbdApiData(sab_api, name, config.get(CONF_SENSORS, {}))
+
+    if config.get(CONF_SENSORS):
+        hass.data[DATA_SABNZBD] = sab_api_data
+        hass.async_add_job(
+            discovery.async_load_platform(hass, 'sensor', DOMAIN, {}, config))
+
+    async def async_service_handler(service):
+        """Handle service calls."""
+        if service.service == SERVICE_PAUSE:
+            await sab_api_data.async_pause_queue()
+        elif service.service == SERVICE_RESUME:
+            await sab_api_data.async_resume_queue()
+        elif service.service == SERVICE_SET_SPEED:
+            speed = service.data.get(ATTR_SPEED, None)
+            if speed is None:
+                speed = '100'
+            await sab_api_data.async_set_queue_speed(speed)
+
+    hass.services.async_register(DOMAIN, SERVICE_PAUSE,
+                                 async_service_handler,
+                                 schema=vol.Schema({}))
+
+    hass.services.async_register(DOMAIN, SERVICE_RESUME,
+                                 async_service_handler,
+                                 schema=vol.Schema({}))
+
+    hass.services.async_register(DOMAIN, SERVICE_SET_SPEED,
+                                 async_service_handler,
+                                 schema=SPEED_LIMIT_SCHEMA)
+
+    async def async_update_sabnzbd(now):
+        """Refresh SABnzbd queue data."""
+        from pysabnzbd import SabnzbdApiException
+        try:
+            await sab_api.refresh_data()
+            async_dispatcher_send(hass, SIGNAL_SABNZBD_UPDATED, None)
+        except SabnzbdApiException as err:
+            _LOGGER.error(err)
+
+    async_track_time_interval(hass, async_update_sabnzbd, UPDATE_INTERVAL)
+
+
+@callback
+def async_request_configuration(hass, config, host):
+    """Request configuration steps from the user."""
+    from pysabnzbd import SabnzbdApi
+
+    configurator = hass.components.configurator
+    # We got an error if this method is called while we are configuring
+    if host in _CONFIGURING:
+        configurator.async_notify_errors(
+            _CONFIGURING[host],
+            'Failed to register, please try again.')
+
+        return
+
+    async def async_configuration_callback(data):
+        """Handle configuration changes."""
+        api_key = data.get(CONF_API_KEY)
+        sab_api = SabnzbdApi(host, api_key)
+        if await async_check_sabnzbd(sab_api):
```

Comment:
Invert this check and return if it's true, to make it a guard clause. Then you can outdent the following code.